### ISSUE TITLE
frame: error reporting considered futile

### ISSUE THREAD

**Issue by mmatczuk** [mmatczuk/scylla-go-driver#98]
After a closer examination I think that the buffer error reporting is useless.

Write path - it will never happen that we send invalid data.
```
WriteConsistency
		b.recordError(fmt.Errorf("invalid consistency: %v", v))
WriteEventTypes
			b.recordError(fmt.Errorf("invalid EventType %s", k))
WriteInet
		b.recordError(fmt.Errorf("invalid IP length"))
WriteOpCode
		b.recordError(fmt.Errorf("invalid operation code: %v", v))
WriteStartupOptions
			b.recordError(fmt.Errorf("invalid mandatory Startup option %s: %s", k, s))
			b.recordError(fmt.Errorf("invalid Startup option %s: %s", k, s))
		b.recordError(fmt.Errorf("invalid Startup option"))
WriteValue
		b.recordError(fmt.Errorf("invalid value"))
```

To save this work for future I suggest adding `frame.Debug` variable.
It the flag is on we check the condition, if condition is met we log it.

Read path - premature handling of unsupported options.

```
ReadConsistency
		b.recordError(fmt.Errorf("invalid consistency: %v", v))
ReadErrorCode
		b.recordError(fmt.Errorf("invalid error code: %d", v))
ReadInet
	b.recordError(fmt.Errorf("invalid ip length"))
ReadOpCode
		b.recordError(fmt.Errorf("invalid operation code: %v", o))
ReadOption
			b.recordError(fmt.Errorf("invalid Option ID: %d", id))
ReadSchemaChangeTarget
		b.recordError(fmt.Errorf("invalid SchemaChangeTarget: %s", v))
ReadSchemaChangeType
		b.recordError(fmt.Errorf("invalid SchemaChangeType: %s", t))
ReadStatusChangeType
		b.recordError(fmt.Errorf("invalid StatusChangeType: %s", t))
ReadTopologyChangeType
		b.recordError(fmt.Errorf("invalid TopologyChangeType: %s", t))
ReadValue
		b.recordError(fmt.Errorf("invalid value length"))
ReadWriteType
		b.recordError(fmt.Errorf("invalid write type: %s", w))
```

If server sent OpCode we do not support the place to handle that is in the main parsing function.
The fact that constant exists does not guarantee that it is supported anyways.
Checking for those errors actually is mixed up with i/o errors wich is unfortunate.
I suggest we do the same as with Write path - use `frame.Debug` variable.

Read - handling errors that would never happen

```
read
		b.recordError(fmt.Errorf("buffer read error: %w", err))
		b.recordError(fmt.Errorf("buffer read error: invalid length"))
```

We can assume that server sends valid data, and buffer always contains all the data needed to parse.
This can be a panic().

---

To summarize:
- Introduce `frame.Debug` variable and only do the checks if the flag is true
- Remove Buffer error
- In read add panics if we cannot read enough data
